exporters/otlp/src/main/java/io/opentelemetry/exporters/otlp/OtlpGrpcMetricExporter.java

```diff
@@ -89,26 +97,29 @@ private OtlpGrpcMetricExporter(ManagedChannel channel, long deadlineMs) {
    * @return the result of the operation
    */
   @Override
-  public ResultCode export(Collection<MetricData> metrics) {
+  public CompletableResultCode export(Collection<MetricData> metrics) {
     ExportMetricsServiceRequest exportMetricsServiceRequest =
         ExportMetricsServiceRequest.newBuilder()
             .addAllResourceMetrics(MetricAdapter.toProtoResourceMetrics(metrics))
             .build();
 
-    try {
-      MetricsServiceGrpc.MetricsServiceBlockingStub stub = this.blockingStub;
-      if (deadlineMs > 0) {
-        stub = stub.withDeadlineAfter(deadlineMs, TimeUnit.MILLISECONDS);
-      }
+    final CompletableResultCode result = new CompletableResultCode();
+    Futures.addCallback(
+        metricsService.export(exportMetricsServiceRequest),
+        new FutureCallback<ExportMetricsServiceResponse>() {
+          @Override
+          public void onSuccess(@Nullable ExportMetricsServiceResponse response) {
+            result.succeed();
+          }
 
-      // for now, there's nothing to check in the response object
-      // noinspection ResultOfMethodCallIgnored
-      stub.export(exportMetricsServiceRequest);
-      return ResultCode.SUCCESS;
-    } catch (Throwable e) {
-      logger.log(Level.WARNING, "Failed to export metrics", e);
-      return ResultCode.FAILURE;
-    }
+          @Override
+          public void onFailure(Throwable t) {
+            logger.log(Level.WARNING, "Failed to export spans", t);
+            result.fail();
```

Comment:
This does make me think we should consider adding an optional Throwable/message to the `fail()` method.

Reply:
I think there's some trade-off (a higher chance if double logging) bug either seems fine for me, including following up in another PR since it's not an API break.

Currently this PR seems to have been blocked by 

https://github.com/open-telemetry/opentelemetry-java/pull/1428#issuecomment-667714059

with no real action item.